cmd/nodepool/core/create_test.go

```diff
@@ -102,3 +107,106 @@ func TestValidateHostedClusterPayloadSupportsNodePoolCPUArch(t *testing.T) {
 		})
 	}
 }
+
+func TestValidMinorVersionCompatibility(t *testing.T) {
+	log.SetLogger(zap.New(zap.UseDevMode(true)))
+
+	tests := []struct {
+		name                 string
+		controlPlaneVersion  string
+		nodePoolReleaseImage string
+		expectedError        string
+	}{
+		{
+			name:                 "same version",
```

Comment:
nit: usually prefer "when ... it should ..." format

Reply:
done